tests/e2e/upgrade/node.go

```diff
@@ -0,0 +1,41 @@
+package upgrade
+
+import "github.com/ory/dockertest/v3"
+
+var baseCmd = []string{"evmosd", "start"}
+
+type Node struct {
+	repository string
+	version    string
+
+	runOptions     *dockertest.RunOptions
+	withRunOptions bool
+}
+
+func NewNode(repository, version string) *Node {
+	return &Node{
+		repository: repository,
+		version:    version,
+		runOptions: &dockertest.RunOptions{
+			Repository: repository,
+			Tag:        version,
+			Cmd:        baseCmd,
+		},
+	}
+}
+
+func (n *Node) Mount(mountPath string) *Node {
+	n.runOptions.Mounts = []string{mountPath}
+	n.UseRunOptions()
+	return n
+}
+
+func (n *Node) Cmd(cmd []string) *Node {
```

Comment:
no need to return if you're using the pointer. We can also rename this to
```suggestion
func (n *Node) SetCmd(cmd []string) {
```

Reply:
Thought to create setters chain like
```
node := NewNode(repo, version).SetCmd(cmd).Mount(path) ...
```

Reply:
> no need to return if you're using the pointer

Thought to create setters chain like
```
node := NewNode(repo, version).SetCmd(cmd).Mount(path) ...
```

Reply:
you can also use
```go
node := NewNode(repo, version)
node.SetCmd(cmd).Mount(path)
```